WordPress/Classes/ViewRelated/Menus/MenuItemEditingViewController.m

```diff
@@ -0,0 +1,470 @@
+        self.headerView.alpha = hidden ? 0.0 : 1.0;
+        if (!hidden) {
+            [self.headerView setNeedsDisplay];
+        }
+    }
+}
+
+- (void)updateLayoutIfNeeded
+{
+    // compactWidthLayout is any screen size in which the width is less than the height (iPhone portrait)
+    BOOL compactWidthLayout = [self shouldLayoutForCompactWidth];
+    BOOL minimizeLayoutForSourceViewTypying = !IS_IPAD && self.sourceViewIsTyping;
+    
+    if (minimizeLayoutForSourceViewTypying) {
+        // headerView should be hidden while typing within the sourceView, to save screen space (iPhone)
+        [self setHeaderViewHidden:YES];
+    } else  {
+        [self setHeaderViewHidden:NO];
+    }
+    
+    if (!IS_IPAD) {
+        
+        if (!compactWidthLayout) {
+            // on iPhone landscape we want to minimize the height of the footer to gain any vertical screen space we can
+            self.footerViewHeightConstraint.constant = MenuItemEditingFooterViewCompactHeight;
+        } else  {
+            // restore the height of the footer on portrait since we have more vertical screen space
+            self.footerViewHeightConstraint.constant = MenuItemEditingFooterViewDefaultHeight;
+        }
+    }
+    
+    if (compactWidthLayout || minimizeLayoutForSourceViewTypying) {
+        
+        [self setContentLayout:MenuItemEditingViewControllerContentLayoutDisplaysSourceView];
+        
+    } else  {
+        
+        [self setContentLayout:MenuItemEditingViewControllerContentLayoutDisplaysTypeAndSourceViews];
+    }
+
+    [self.stackView layoutIfNeeded];
+    [self setNeedsStatusBarAppearanceUpdate];
+    [self.headerView setNeedsTopConstraintsUpdateForStatusBarAppearence:self.headerView.hidden];
+}
+
+- (void)updateLayoutIfNeededAnimated
+{
+    [self.stackView layoutIfNeeded];
+    [UIView animateWithDuration:0.20 animations:^{
+        [self updateLayoutIfNeeded];
+    }];
+}
+
+- (void)transitionLayoutToDisplayTypeView
+{
+    [self setContentLayout:MenuItemEditingViewControllerContentLayoutDisplaysTypeView];
+    [UIView animateWithDuration:0.15 delay:0.0 options:UIViewAnimationOptionCurveEaseInOut animations:^{
+        
+        [self.typeViewController updateDesignForLayoutChangeIfNeeded];
+        [self.contentView layoutIfNeeded];
+        
+    } completion:nil];
+}
+
+- (void)transitionLayoutToDisplaySourceView
+{
+    if ([self shouldLayoutForCompactWidth]) {
+        [self setContentLayout:MenuItemEditingViewControllerContentLayoutDisplaysSourceView];
+    } else  {
+        [self setContentLayout:MenuItemEditingViewControllerContentLayoutDisplaysTypeAndSourceViews];
+    }
+    [UIView animateWithDuration:0.15 delay:0.0 options:UIViewAnimationOptionCurveEaseInOut animations:^{
+        
+        [self.contentView layoutIfNeeded];
+        
+    } completion:nil];
+}
+
+- (void)setContentLayout:(MenuItemEditingViewControllerContentLayout)contentLayout
+{
+    if (_contentLayout != contentLayout) {
```

Comment:
super nitpicky.  maybe check for equality and return early to save a level of nesting.